thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -2011,19 +2009,15 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor
     void readAhead()
     {
         endOfInput = false;
-        CJoinGroup *lastGroupMember = nullptr;
+        CJoinGroup *groupStart = nullptr;
         do
         {
             if (queryAbortSoon())
                 break;
             OwnedConstThorRow lhsRow = inputStream->nextRow();
             if (!lhsRow)
             {
-                if (preserveGroups && lastGroupMember)
-                {
-                    lastGroupMember->addFlag(GroupFlags::gf_eog);
-                    lastGroupMember = nullptr;
-                }
```

Comment:
Removing this, and setting gf_start (renamed from gf_eof) on line 2047 is the actual fix.

The rest if tidying up, improving comments slightly.